### PR TITLE
Pass changes in cfd proposals into projection actor

### PR DESCRIPTION
**Pull request by klochowicz**
Instead of passing the whole HashMap to the projection actor, only send what has
actually changed.

Provide separate messages to update settlement proposal and roll over proposals,
as they will be sent from different actors.

Some type conversions added in this commit are meant to be temporary and will disappear when cfd actors will not be storing a joint hashmap of proposals (both settlement and rollover) anymore.

### CODE REVIEW

File: daemon/src/taker_cfd.rs [hunk 1]
```diff
@@ -162,14 +179,13 @@ where
...
-                proposal: proposal.clone(),
-                direction: SettlementKind::Outgoing,
-            },
-        );
-        self.send_pending_update_proposals().await?;
+        let new_proposal = UpdateCfdProposal::RollOverProposal {
+            proposal: proposal.clone(),
+            direction: SettlementKind::Outgoing,
+        };
+        self.projection_actor
```
Comment (thomaseizinger) [inline daemon/src/taker_cfd.rs:189]: Are we on purpose not inserting here?

**Review by thomaseizinger — APPROVED** [review]
Thanks!

File: daemon/src/taker_cfd.rs (hunk 1 above)
Comment (klochowicz) [reply daemon/src/taker_cfd.rs:189]: oops, I missed that one! thanks for spotting! added the missing `insert`

**Pull request merged**